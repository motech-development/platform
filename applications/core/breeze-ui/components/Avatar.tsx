import { FC } from 'react';

const Avatar: FC<{
  alt: string;
  src: string;
  width?: number | 'auto';
}> = ({ alt, src, width = 'auto' }) => (
  <img className={`h-${width} w-${width} rounded-full`} src={src} alt={alt} />
);

export default Avatar;
