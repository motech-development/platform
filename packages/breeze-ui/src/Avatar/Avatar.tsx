import React, { FC, memo } from 'react';
import styled from 'styled-components';

interface IImageProps {
  $width: number;
}

const Image = styled.img<IImageProps>`
  ${({ $width }) => `
    display: block;
    width: ${$width ? `${$width}px` : 'auto'};
  `}
`;

interface IWrapperProps {
  $width: number;
}

const Wrapper = styled.div<IWrapperProps>`
  ${({ $width }) => `
    border-radius: 50%;
    display: inline-block;
    overflow: hidden;
    width: ${$width ? `${$width}px` : 'auto'};
  `}
`;

export interface IAvatarProps {
  alt: string;
  src: string;
  width?: number;
}

const Avatar: FC<IAvatarProps> = ({ alt, src, width = 0 }) => (
  <Wrapper $width={width}>
    <Image alt={alt} src={src} $width={width} />
  </Wrapper>
);

export default memo(Avatar);
