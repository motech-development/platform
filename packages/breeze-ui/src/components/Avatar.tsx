import { DetailedHTMLProps, FC, ImgHTMLAttributes } from 'react';
import { classNames } from '../utils/className';

export interface IAvatarProps
  extends DetailedHTMLProps<
    ImgHTMLAttributes<HTMLImageElement>,
    HTMLImageElement
  > {
  alt: string;
  src: string;
}

const Avatar: FC<IAvatarProps> = ({
  alt,
  className = 'h-8 w-8',
  src,
  ...rest
}) => (
  <img
    className={classNames('rounded-full', className)}
    src={src}
    alt={alt}
    {...rest}
  />
);

export default Avatar;
