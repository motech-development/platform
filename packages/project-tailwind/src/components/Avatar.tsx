import { ComponentPropsWithRef } from 'react';

/** Avatar component properties */
export interface IAvatarProps extends ComponentPropsWithRef<'img'> {
  /** Alternative information text */
  alt: string;

  /** Image path */
  src: string;
}

/**
 * Display user avatar photo
 *
 * @param props - Component props
 *
 * @returns Avatar component
 */
export function Avatar(props: IAvatarProps) {
  const { alt, className = 'h-8 w-8', src, ...rest } = props;

  return <img {...{ alt, className, src }} {...rest} />;
}
