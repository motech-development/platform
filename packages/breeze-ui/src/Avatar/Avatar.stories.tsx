import { withA11y } from '@storybook/addon-a11y';
import { number, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import BaseStyles from '../BaseStyles/BaseStyles';
import Avatar from './Avatar';

const stories = storiesOf('Avatar', module);

stories.addDecorator(withA11y);
stories.addDecorator(withKnobs);

stories.add('Basic avatar', () => (
  <>
    <BaseStyles />
    <Avatar
      alt={text('Alt text', 'Avatar')}
      src={text(
        'Image Url',
        'https://www.gravatar.com/avatar/8801091e665fdac669daa63d32167b7b',
      )}
      width={number('Width', 0)}
    />
  </>
));
