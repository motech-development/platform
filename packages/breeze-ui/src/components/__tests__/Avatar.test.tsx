import { render } from '@testing-library/react';
import Avatar from '../Avatar';

describe('Avatar', () => {
  it('should correctly render component when no styles are set', () => {
    const { container } = render(
      <Avatar
        alt="Alt text"
        src="https://www.gravatar.com/avatar/8801091e665fdac669daa63d32167b7b"
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should correctly render component when custom styles are set', () => {
    const { container } = render(
      <Avatar
        className="w-auto h-auto"
        alt="Alt text"
        src="https://www.gravatar.com/avatar/8801091e665fdac669daa63d32167b7b"
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
