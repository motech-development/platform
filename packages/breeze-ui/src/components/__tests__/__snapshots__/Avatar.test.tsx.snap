// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should correctly render component when custom styles are set 1`] = `
<img
  alt="Alt text"
  class="rounded-full w-auto h-auto"
  src="https://www.gravatar.com/avatar/8801091e665fdac669daa63d32167b7b"
/>
`;

exports[`Avatar should correctly render component when no styles are set 1`] = `
<img
  alt="Alt text"
  class="rounded-full h-8 w-8"
  src="https://www.gravatar.com/avatar/8801091e665fdac669daa63d32167b7b"
/>
`;
